x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/response/voyageai/VoyageAIEmbeddingsResponseEntity.java

```diff
@@ -0,0 +1,201 @@
+            } else if (embeddingType == VoyageAIEmbeddingType.INT8) {
+                List<InferenceByteEmbedding> embeddingList = parseList(
+                    jsonParser,
+                    VoyageAIEmbeddingsResponseEntity::parseEmbeddingObjectByte
+                );
+
+                return new InferenceTextEmbeddingByteResults(embeddingList);
+            } else if (embeddingType == VoyageAIEmbeddingType.BIT || embeddingType == VoyageAIEmbeddingType.BINARY) {
+                List<InferenceByteEmbedding> embeddingList = parseList(
+                    jsonParser,
+                    VoyageAIEmbeddingsResponseEntity::parseEmbeddingObjectBit
+                );
+
+                return new InferenceTextEmbeddingBitResults(embeddingList);
+            } else {
+                throw new IllegalArgumentException(
+                    "Illegal embedding_type value: " + embeddingType + ". Supported types are: " + VALID_EMBEDDING_TYPES_STRING
+                );
+            }
+        }
+    }
+
+    private static InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding parseEmbeddingObjectFloat(XContentParser parser)
+        throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+
+        positionParserAtTokenAfterField(parser, "embedding", FAILED_TO_FIND_FIELD_TEMPLATE);
+
+        List<Float> embeddingValuesList = parseList(parser, XContentUtils::parseFloat);
+        // parse and discard the rest of the object
+        consumeUntilObjectEnd(parser);
+
+        return InferenceTextEmbeddingFloatResults.InferenceFloatEmbedding.of(embeddingValuesList);
+    }
+
+    private static InferenceByteEmbedding parseEmbeddingObjectByte(XContentParser parser) throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+
+        positionParserAtTokenAfterField(parser, "embedding", FAILED_TO_FIND_FIELD_TEMPLATE);
+
+        List<Byte> embeddingValuesList = parseList(parser, VoyageAIEmbeddingsResponseEntity::parseEmbeddingInt8Entry);
+        // parse and discard the rest of the object
+        consumeUntilObjectEnd(parser);
+
+        return InferenceByteEmbedding.of(embeddingValuesList);
+    }
+
+    private static InferenceByteEmbedding parseEmbeddingObjectBit(XContentParser parser) throws IOException {
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, parser.currentToken(), parser);
+
+        positionParserAtTokenAfterField(parser, "embedding", FAILED_TO_FIND_FIELD_TEMPLATE);
+
+        List<Byte> embeddingValuesList = parseList(parser, VoyageAIEmbeddingsResponseEntity::parseEmbeddingBitEntry);
+        // parse and discard the rest of the object
+        consumeUntilObjectEnd(parser);
+
+        return InferenceByteEmbedding.of(embeddingValuesList);
+    }
+
+    private static Byte parseEmbeddingInt8Entry(XContentParser parser) throws IOException {
```

Comment:
Looks like this might be a duplicate of the function below: `parseEmbeddingBitEntry`, how about we remove one of them.